New `UnusedDocParam` theme-check rule (#808)
Shopify/theme-tools: .changeset/dull-badgers-sing.md

---
'@shopify/theme-check-common': minor
'@shopify/theme-check-node': minor
---

New `UnusedDocParam` theme-check rule

> - Theme check will verify if variable defined in `@param` tag within a snippet's `doc` is used

Reviewer: ```suggestion
- Theme check will verify that parameters defined within a snippet's `doc` header are used within the body
```
